qiskit/circuit/parametertable.py

```diff
@@ -12,43 +12,113 @@
 """
 Look-up table for variable parameters in QuantumCircuit.
 """
-import warnings
 import functools
-from collections.abc import MutableMapping, MappingView
+import warnings
+from collections.abc import MappingView, MutableMapping, MutableSet
+
+
+class ParameterReferences(MutableSet):
+    """A set of instruction parameter slot references.
+    Items are expected in the form ``(instruction, param_index)``. Membership
+    testing is overriden such that items that are otherwise value-wise equal
+    are still considered distinct if their ``instruction``\\ s are referentially
+    distinct.
+    """
+
+    def _instance_key(self, ref):
+        return (id(ref[0]), ref[1])
+
+    def __init__(self, refs):
+        self._instance_ids = {}
+
+        for ref in refs:
+            if not isinstance(ref, tuple) or len(ref) != 2:
+                raise ValueError("refs must be in form (instruction, param_index)")
+            k = self._instance_key(ref)
+            self._instance_ids[k] = ref[0]
+
+    def __len__(self):
+        return len(self._instance_ids)
+
+    def __iter__(self):
+        for (_, idx), instruction in self._instance_ids.items():
+            yield (instruction, idx)
+
+    def __contains__(self, x) -> bool:
+        return self._instance_key(x) in self._instance_ids
+
+    def __repr__(self) -> str:
+        return f"ParameterReferences({repr(list(iter(self)))})"
```

Comment:
Incredibly minor, but `list(self)` should already do the right thing - no need to manually call `iter`.

Reply:
Fixed in 1ff03a5.